### ISSUE TITLE
【Not an issue】A quick start to use SAM2 interactively

### ISSUE THREAD

**Issue by lujiazho** [facebookresearch/segment-anything-2#73]
I developed this simple [tool](https://github.com/lujiazho/SegDrawer) since the release of SAM. I just naively integrated the newly released SAM2 into it. Feel free to give it a try or adapt it to your own application! Feel free to contribute and fix some bugs as well.

<table>
  <tr>
    <td align="center">
      <img src="https://raw.githubusercontent.com/lujiazho/SegDrawer/main/example/video_seg/BostonDynamics_ori.gif" width="200" />
    </td>
    <td align="center">
      first frame seg<br>
      <img src="https://raw.githubusercontent.com/lujiazho/SegDrawer/main/example/video_seg/BostonDynamics_mask.png" width="200" />
    </td>
    <td align="center">
      SAM2<br>
      <img src="https://raw.githubusercontent.com/lujiazho/SegDrawer/main/example/video_seg/BostonDynamics_sam2_seg.gif" width="200" />
    </td>
  </tr>
</table>

**Comment by patermars** [comment]
Wow, So quick, Thanks for this. Really needed something like that.

**Comment by emi-dm** [comment]
@lujiazho How you make it work with CPU only? Can you use SAM-2 with only CPU?

**Comment by lujiazho** [comment]
> @lujiazho How you make it work with CPU only? Can you use SAM-2 with only CPU?

Yes we can. Just use a machine with only CPU or manually set `device='cpu'`. Try it out on colab! It would be crazily slow though.

**Comment by songhat** [comment]
> > @lujiazho How you make it work with CPU only? Can you use SAM-2 with only CPU?
> 
> Yes we can. Just use a machine with only CPU or manually set `device='cpu'`. Try it out on colab! It would be crazily slow though.

can this tool eazily put any frame into the SAM2, or just only can use first frame?
